### PR TITLE
Controllers 👨‍💻 ⚡ 

### PR DESCRIPTION
**Pull request by tsheporamantso**
### On this milestone 🗿 I've implemented the following 🧠 
- [x] Add nested resourceful routing for `users` and `posts` with `index` and `show` actions respectively ✔️ 
- [x] Generate `users_controller` and `posts_controller` with `rails g controller --skip-routes` command ✔️
- [x] Define `index` action that returns all users and all posts and also `show` action that finds user and post by id parameter ✔️ 
- [x]  Generate `posts` and `users`  views folder with placeholder text respectively ✔️ 
- [x] Check and correct `rubocop` offenses 🤖 
     - Please see below snapshot of rails routes     
image

### CODE REVIEW

File: spec/helpers/posts_helper_spec.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+# describe PostsHelper do
+#   describe "string concat" do
+#     it "concats two strings with spaces" do
+#       expect(helper.concat_strings("this","that")).to eq("this that")
+#     end
+#   end
+# end
+RSpec.describe PostsHelper, type: :helper do
+  pending "add some examples to (or delete) #{__FILE__}"
+end
```
Comment (TedLivist) [inline spec/helpers/posts_helper_spec.rb:15]: - [ ] Kindly remove this generated file. You were required not to use generate command.

File: spec/helpers/users_helper_spec.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+# describe UsersHelper do
+#   describe "string concat" do
+#     it "concats two strings with spaces" do
+#       expect(helper.concat_strings("this","that")).to eq("this that")
+#     end
+#   end
+# end
+RSpec.describe UsersHelper, type: :helper do
+  pending "add some examples to (or delete) #{__FILE__}"
+end
```
Comment (TedLivist) [inline spec/helpers/users_helper_spec.rb:15]: - [ ] Kindly remove this generated file. You were required not to use generate command.

File: spec/views/posts/index.html.erb_spec.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe 'posts/index.html.erb', type: :view do
+  pending "add some examples to (or delete) #{__FILE__}"
+end
```
Comment (TedLivist) [inline spec/views/posts/index.html.erb_spec.rb:5]: - [ ] Kindly remove this generated file. You were required not to use generate command.

File: app/helpers/posts_helper.rb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+module PostsHelper
```
Comment (TedLivist) [inline app/helpers/posts_helper.rb:1]: - [ ] Kindly remove this generated file. You were required not to use generate command.

File: app/helpers/users_helper.rb [hunk 5]
```diff
@@ -0,0 +1,2 @@
+module UsersHelper
+end
```
Comment (TedLivist) [inline app/helpers/users_helper.rb:2]: - [ ] Kindly remove this generated file. You were required not to use generate command.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @tsheporamantso,

Good job so far!
✔️ Implementation nested routes
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/views/posts/index.html.erb_spec.rb (hunk 3 above)
Comment (tsheporamantso) [reply spec/views/posts/index.html.erb_spec.rb:5]: Good day @TedLivist,
I appreciate your review. However, I beg to differ. I was required not to use `rails generate scaffhold` and that's not what I used. I used `rails generate controller`
Can you please reconsider the review.

Regards,
Gladwin.

**Comment by bellom** [comment]
Hello @tsheporamantso 👋, 

This is an additional review as a result of the second opinion request. 

Thank you for reaching out to us. 

According to the [project requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Setup_Controllers_project_v1_1.md#:~:text=The%20point%20of%20this%20project%20is%20to%20show%20you%20how%20to%20set%20up%20controllers%20by%20yourself.), you are supposed to create the controller by yourself without using any generating command, see here 👇 
image

### Requested Changes ♻️ 
Kindly attend to all the requested changes from @TedLivist.

[Optional]
It would be nice to redo this milestone so that you can learn the lesson objectives. I made these changes as optional suggestions because the CR didn't require you to redo the work. He was supposed to have 

Keep rocking 🚀!

**Comment by bellom** [comment]
@tsheporamantso - This is under re-review.

**Comment by tsheporamantso** [comment]
> @tsheporamantso - This is under re-review.

**Comment by bellom** [comment]
Hello @tsheporamantso 👋, 

This is an additional review as a result of the second opinion request. 

After a deep look at this case, you can use `rails g controller`, since the command will create an empty controller file and you will have to manually add the actions in the controller file.

Sorry for the mix-up 🙏 

As this was the only reason your project was not approved, it will be approved as soon as possible.

Keep rocking 🚀!

**Pull request merged**